compendium/DeclarativeServices/src/manager/ReferenceManagerImpl.cpp

```diff
@@ -384,5 +267,35 @@ void ReferenceManagerImpl::BatchNotifyAllListeners(const std::vector<RefChangeNo
     }
   }
 }
+
+// util method to extract service-id from a given reference
+long ReferenceManagerBaseImpl::GetServiceId(const ServiceReferenceBase& sRef)
+{
+  auto idAny = sRef.GetProperty(cppmicroservices::Constants::SERVICE_ID);
+  return cppmicroservices::any_cast<long>(idAny);
+}
+
+std::unique_ptr<ReferenceManagerBaseImpl::BindingPolicy> ReferenceManagerBaseImpl::CreateBindingPolicy(ReferenceManagerBaseImpl& ref
+                                                                                                       , const std::string& policy
+                                                                                                       , const std::string& policyOption)
+{
+  if (policy == "static") {
+    if (policyOption == "reluctant") {
+      return std::make_unique<BindingPolicyStaticReluctant>(ref);
+    }
+    else { // greedy
+      return std::make_unique<BindingPolicyStaticGreedy>(ref);
+    }
+  }
+  else { // dynamic
+    if (policyOption == "reluctant") {
+      return std::make_unique<BindingPolicyDynamicReluctant>(ref);
+    }
+    else { // greedy
+      return std::make_unique<BindingPolicyDynamicGreedy>(ref);
+    } 
+  }
+}
+
 }
 }
```

Comment:
minor detail: are you planning to change the file name to ReferenceManagerBaseImpl?

Reply:
nope, at least not in this PR